docs/user-interaction-config-file-spec.rst

```diff
@@ -0,0 +1,204 @@
+has changed an option on the screen. If the option is changed by the user, the corresponding key is set
+to 1. If the given option has not been changed by the user then the corresponding key can either be
+omitted or set to 0.
+
+Example:
+
+::
+
+    [DatetimeSpoke]
+    visited=1
+    changed_timezone=1
+    changed_ntp=0
+    changed_timedate=1
+
+In this example the user has visited the date & time spoke and has changed the timezone & time/date,
+but not the NTP settings. Note that the ``changed_ntp`` key could also be omitted as the user has not changed
+the NTP options.
+
+Another example:
+
+::
+
+    [KeyboardSpoke]
+    visited=0
+
+Here the user has not visited the keyboard spoke and thus could not have changed any options,
+so all ``changed_*`` keys (if any) have been omitted.
+
+Note that if a spoke section is missing, it should be assumed that the corresponding screen has not been visited.
+On othe other hand, if a screen *has been visited*, the section *must* be present, with the ``visited`` key being equal to 1.
+
+
+Full configuration file example
+-------------------------------
+
+::
+
+    post_install_tools_disabled=0
+
+    [DatetimeSpoke]
+    visited=1
+    changed_timezone=1
+    changed_ntp=0
+    changed_timedate=1
+
+    [KeyboardSpoke]
+    visited=0
+
+On the first line of the example the ``post_install_tools_disabled`` is equal to 0, which means that
+post installation setup tools should proceed as usual. In this case the ``post_install_tools_disabled`` key
+might also be omitted.
+
+Next there are two sections corresponding to two screens - ``Datetimespoke`` and ``KeyboardSpoke``.
+
+The user has visited the date & time screen and has changed various options, but not the NTP settings.
+On the other hand the keyboard screen has not been visited at all.
+
+
+Parsing and writing the of the configuration file by Anaconda
+-------------------------------------------------------------
+
+If the user interaction file exists during Anaconda startup, it will be parsed and taken into account
+when deciding which screens to show during the installation. This make it possible for secondary
+installation setup tools to run before Anaconda and query the user for information.
+
+This can be for example a tool querying the user for language settings. Then once Anaconda starts it can
+skip the language selection screen as language has already been set by the tool.
+
+Once the installation process is done Anaconda will write out information about what screens the user has
+and has not visited and optionally which settings have been changed by the user.
+
+If Anaconda successfully parsed an existing user interaction configuration file, any valid settings present
+in the file will propagate to the configuration file when it is written-out by Anaconda.
+
+Note that comments present in the configuration file at the time Anaconda parses it might not be present
```

Comment:
The python configparser module takes both "# "and ";", so lets go with "#" so that GKeyFile can be used - an I guess I'll add GKeyFile to the list of suggested parser for the format when I am at it.

Also lets disallow inline comments - they don't seem to be supported very well across parsers, configparser does not parse them by default & advises against using them.

So this would be valid:

    # comment
    [Foo]
    # comment
    bar=baz
    # comment

But this not:

    [Foo]
    bar=baz # comment